### PR TITLE
switch mnist dataset to datapythonista/mnist

### PR DESCRIPTION
**Pull request by npapernot**
A recent change to the TF API creates a deprecation warning because CleverHans uses the MNIST dataset loader that comes with `tf.learn`.

This implements the suggested switch to `sklearn` as the data loader.

### CODE REVIEW

**Comment by npapernot** [comment]
Partially addresses #397

**Comment by nottombrown** [comment]
Added the missing `scikit-learn` dependency.

**Comment by nottombrown** [comment]
I give permission to include my commits in this PR

**Comment by nottombrown** [comment]
Found an issue here. The sklearn data is sorted by categories, so our train and test set are not IID. 

Two options:
1. Shuffle the mnist data ourselves
2. Use MNIST dataset from `torchvision.datasets` (https://pytorch.org/docs/stable/torchvision/datasets.html#mnist)

I would gravitate towards [2] because torchvision seems lighter-weight than sklearn, and we don't have shuffle things ourselves.

**Comment by nottombrown** [comment]
Heavyweight pytorch dependency is causing issues, so I switched to a new mnist source.

File: cleverhans/utils_mnist.py [hunk 1]
```diff
@@ -27,18 +25,27 @@ def data_mnist(datadir='/tmp/', train_start=0, train_end=60000, test_start=0,
...
+
+    X_train = mnist.train_images() / 255.
+    Y_train = mnist.train_labels()
+    X_test = mnist.test_images() / 255.
+    Y_test = mnist.test_labels()
+
+    X_train = np.expand_dims(X_train, -1)
+    X_test = np.expand_dims(X_test, -1)
+
+    print(X_train.max())
```
Comment (npapernot) [inline cleverhans/utils_mnist.py]: We can probably drop these two statements to save terminal real estate

**Comment by npapernot** [comment]
Does the new loader return images in a different random order every run?

**Comment by nottombrown** [comment]
Removed print statements. The order of the loader is deterministic. It uses the ordering from http://yann.lecun.com/exdb/mnist/

You can see the code here:
https://github.com/datapythonista/mnist/blob/master/mnist/__init__.py

**Comment by npapernot** [comment]
Thanks! LGTM, feel free to merge when tests pass
